src/PrestaShopBundle/Controller/Admin/Improve/International/StateController.php

```diff
@@ -58,10 +69,174 @@ public function getStatesAction(Request $request)
+        }
+
+        return $this->render('@PrestaShop/Admin/Improve/International/Locations/State/add.html.twig', [
+            'stateForm' => $attachmentForm->createView(),
+            'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
+        ]);
+    }
+
+    /**
+     * Handles edit form rendering and submission
+     *
+     * @AdminSecurity(
+     *     "is_granted('update', request.get('_legacy_controller'))",
+     *     redirectRoute="admin_states_index"
+     * )
+     * @DemoRestricted(redirectRoute="admin_states_index")
+     *
+     * @param int $stateId
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function editAction(int $stateId, Request $request): Response
+    {
+        try {
+            /** @var EditableState $stateInformation */
+            $stateInformation = $this->getQueryBus()->handle(new GetStateForEditing((int) $stateId));
+        } catch (StateException $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages()));
+
+            return $this->redirectToRoute('admin_states_index');
+        }
+
+        $stateForm = null;
+
+        try {
+            $stateFormBuilder = $this->get(
+                'prestashop.core.form.identifiable_object.builder.state_form_builder'
+            );
+
+            $stateFormHandler = $this->get(
+                'prestashop.core.form.identifiable_object.handler.state_form_handler'
+            );
+
+            $stateForm = $stateFormBuilder->getFormFor((int) $stateId);
+            $stateForm->handleRequest($request);
+            $result = $stateFormHandler->handleFor((int) $stateId, $stateForm);
+            if ($result->isSubmitted() && $result->isValid()) {
+                $this->addFlash('success', $this->trans('Successful update.', 'Admin.Notifications.Success'));
+
+                return $this->redirectToRoute('admin_states_index');
+            }
+        } catch (Exception $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages()));
+        }
+
+        return $this->render('@PrestaShop/Admin/Improve/International/Locations/State/edit.html.twig', [
+            'stateForm' => null !== $stateForm ? $stateForm->createView() : null,
+            'stateInformation' => $stateInformation,
+            'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
+        ]);
+    }
+
+    /**
+     * @return array
+     */
+    private function getErrorMessages(): array
+    {
+        return [
+            StateException::class => $this->trans(
+                'Unexpected error occurred.',
+                'Admin.Notifications.Error'
+            ),
+            StateNotFoundException::class => $this->trans(
+                'The object cannot be loaded (or found)',
```

Comment:
If I understand well, the two following error notifications here apply to the "States" tab of the "Locations" section of the back office? Why are we talking about an object to load, it seems to be pretty much adding a state manually or editing one without any loading action?

Reply:
For editing an object it needs to be loaded. I don't think that this will happen in practice but StateNotFoundException could be thrown in editing flow so the message should be here.

Reply:
Okay, so when you create/edit a location in the International part of the back office, it is considered as a loaded object?

Reply:
Only on editing but yes